### ISSUE TITLE
Solution : Support for Generic Smart Plugs 

### ISSUE THREAD

**Issue by varungujjar** [xoseperez/espurna#1919]
![IMG_20190922_175239](https://user-images.githubusercontent.com/4302147/65387327-e8dfe980-dd63-11e9-90cc-39d07471a913.jpg)
![IMG_20190922_180308-01](https://user-images.githubusercontent.com/4302147/65387328-e8dfe980-dd63-11e9-9fdb-8a083c3cd883.jpg)

Just beed on this since 3 days now and finally figured out the pins.. hope this helps someone.. :) successfully flashed firmware with following config 

Also this works with the SONOFF_SV Config  in hardware.h file

#define MANUFACTURER        "ITEAD"
    #define DEVICE              "SONOFF_SV"

    // Buttons
    #define BUTTON1_PIN         1
    #define BUTTON1_MODE        BUTTON_PUSHBUTTON | BUTTON_DEFAULT_HIGH
    #define BUTTON1_RELAY       1

    // Relays
    #define RELAY1_PIN          14
    #define RELAY1_TYPE         RELAY_TYPE_NORMAL

    // LEDs
    #define LED1_PIN            13
    #define LED1_PIN_INVERSE    1

**Comment by wildwiz** [comment]
Looks like it is a Tuya's TYW-E2S ESP8285 WI-FI module, same as I found time ago in the Zhilde power strip about an year ago https://github.com/xoseperez/espurna/wiki/Hardware-Zhilde-ZLD-44EU-W On that page there are some links and some more details on how to program the TYW-E2S

**Comment by varungujjar** [comment]
No it's a esp8266 confirmed and successfully.flashed sonoff SV firmware and working fine

**Comment by varunquartic** [comment]
> Looks like it is a Tuya's TYW-E2S ESP8285 WI-FI module, same as I found time ago in the Zhilde power strip about an year ago https://github.com/xoseperez/espurna/wiki/Hardware-Zhilde-ZLD-44EU-W On that page there are some links and some more details on how to program the TYW-E2S

yes the module is the same.. may be we can document it..  for this plug as well.?

**Comment by ColinShorts** [comment]
The esp8285 is an esp8266 with flash memory built in.

**Comment by varungujjar** [comment]
> The esp8285 is an esp8266 with flash memory built in.

got it just checked the post  that xose posted :)

**Comment by varungujjar** [comment]
> Looks like it is a Tuya's TYW-E2S ESP8285 WI-FI module, same as I found time ago in the Zhilde power strip about an year ago https://github.com/xoseperez/espurna/wiki/Hardware-Zhilde-ZLD-44EU-W On that page there are some links and some more details on how to program the TYW-E2S

maybe we can document this.. ? and add it to the hardware page ?

**Comment by mcspr** [comment]
> maybe we can document this.. ? and add it to the hardware page ?

Please do. You can specify the seller or some board markings. I do see something near the ESP board, photo needs another angle :)
